File: 1850-minimum-length-of-string-after-deleting-similar-ends/minimum-length-of-string-after-deleting-similar-ends.js
var minimumLength = function(s) {
    let i = 0
    let j = s.length - 1

    while (i < j && s[i] == s[j]) {
        let char = s[i]
        while (i <= j && s[i] == char) i += 1
        while (i <= j && s[j] == char) j -= 1
    }    
    return j - i + 1
};

File: 1850-minimum-length-of-string-after-deleting-similar-ends/minimum-length-of-string-after-deleting-similar-ends.py
class Solution:
    def minimumLength(self, s: str) -> int:
        i = 0
        j = len(s) - 1
        
        while (i < j and s[i] == s[j]):
            char = s[i]

            while (i <= j and char == s[i]):
                i += 1

            while (i <= j and char == s[j]):
                j -= 1
            
        return j - i + 1
        